### ISSUE TITLE
Why `comparison-with-itself` only flags comparison with Name

### ISSUE THREAD

**Issue by harupy** [pylint-dev/pylint#10713]
Question about the[ `comparison-with-itself`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/comparison-with-itself.html) rule. Is there a reason this rule only flags comparison with `Name` (e.g. `a == a`)? Is it challenging to extend it to support any comparison where the left hand expression is equal to the right hand expression (e.g. `a.b == a.b`)?

**Comment by Pierre-Sassoulas** [comment]
Yes, probably lack of time and keeping it simple and stupid (the primer would confirm this if someone open a PR).

**Comment by jacobtylerwalls** [comment]
But what about:

```py
class A:
  @property
  def now(self):
      return time.time()

a = A()
a.now == a.now  # False, not redundant
```

**Comment by Pierre-Sassoulas** [comment]
This should definitely not raise :

```python
class A:
  @property
  def now(self):
      return time.time()

a = A()
start = a.now
start == a.now  # False, not redundant
```

But the sample with ``a.now == a.now`` is weird to me, in what situation could it be useful ?

**Comment by jacobtylerwalls** [comment]
It was just a fast quick, example, so you need to imagine other properties that are not idempotent:

```py
@property
def maybe_increments():
  if sometimes:
    self._private_counter += 1
  return self._private_counter

if a.maybe_increments != a.maybe_increments:
  keep_processing()
```

I think this needs a proposal before we bless it.

**Comment by Pierre-Sassoulas** [comment]
Right thank you, still weird code looking wrong, but we could find someone to defend this approach.

So the reason (that we forgot but probably discussed at the time of implementation) is that checking if a class name is a property is costly compared to the benefit and that we favor false négatives over false positive.

I'm still +0.5 for this.

**Comment by jacobtylerwalls** [comment]
I'm unclear what the the proposal is. How are we going to check things other than `Name`s?

**Comment by TheGreatApollyon** [comment]
Implemented in https://github.com/pylint-dev/pylint/pull/11216 — identical attribute accesses (same plain-name receiver, non-property attribute) are now flagged, with property accesses explicitly excluded since they are not idempotent.
